### PR TITLE
CFY-3532 CFY-6735 non root restservice

### PR DESCRIPTION
**Pull request by funkyHat**

### CODE REVIEW

File: components/restservice/config/cloudify-restservice.service [hunk 1]
```diff
@@ -4,11 +4,13 @@ Wants=network-online.target dbus.service
...
 
 [Service]
+User={{ node.properties.service_user }}
+Group={{ node.properties.service_user }}
 TimeoutStartSec=0
 Restart=on-failure
 EnvironmentFile=-/etc/sysconfig/cloudify-restservice
 ExecStart=/bin/sh -c '/opt/manager/env/bin/gunicorn \
-    --pid /var/run/gunicorn.pid \
+    --pid /run/cloudify-restservice/pid \
```
Comment (idanmo) [inline components/restservice/config/cloudify-restservice.service]: why /pid and not .pid?

File: inputs/manager-inputs.yaml [hunk 2]
```diff
@@ -62,6 +62,12 @@ inputs:
     type: integer
     default: 20
 
+  rest_service_system_user:
```
Comment (idanmo) [inline inputs/manager-inputs.yaml]: I'm not sure we want to expose these inputs at the moment.

File: types/manager-types.yaml [hunk 3]
```diff
@@ -636,6 +636,9 @@ node_types:
         description: Number of log files to keep when rotating.
         type: integer
         default: { get_input: rest_service_log_files_backup_count }
+      service_user:
```
Comment (idanmo) [inline types/manager-types.yaml]: this should probably be consistent with other user/group properties... os_user, os_group etc...

File: components/restservice/scripts/configure.py [hunk 4]
```diff
@@ -57,6 +57,13 @@ def _deploy_security_configuration():
         'encoding_min_length': 5
     }
 
+    rest_service_user = ctx.node.properties['service_user']
```
Comment (idanmo) [inline components/restservice/scripts/configure.py]: os_user?

File: components/restservice/config/cloudify-restservice.service (hunk 1 above)
Comment (funkyHat) [inline components/restservice/config/cloudify-restservice.service]: Because `cloudify-restservice` doesn't have permission to create files in `/run`

**Comment by idanmo** [comment]
How about adding an `os_group` property for being consistent with other services?

**Pull request merged**